lib/dry/system/provider_registrar.rb

```diff
@@ -136,6 +136,14 @@ def provider_files
         }.first
       end
 
+      # Extension point for subclasses
+      # @api private
```

Comment:
```suggestion
      # @api public
      # @since 1.1.0
```

I think we'll want to mark these as public since we're using them from Hanami.

(Same thing applies to the other new method below).